src/NServiceBus.Core/Sagas/Sagas.cs

```diff
@@ -43,6 +43,11 @@ protected internal override void Setup(FeatureConfigurationContext context)
                 throw new Exception("The selected persistence doesn't have support for saga storage. Select another persistence or disable the sagas feature using endpointConfiguration.DisableFeature<Sagas>()");
             }
 
+            if (!context.Container.HasComponent<ISagaIdGenerator>())
```

Comment:
Looking at the [persitence definition](https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Persistence/PersistenceDefinition.cs) again I wonder if we could add this there as a small step towards separated persistence definitions for each relevant storage type.

We could have a optional SagaPersistenceDefiniton GetSagaPersistenceDefintion() on the current PersitenceDefinition where we could add this?

That would make us prepared for a future model where we would get that specific definition from a dedicated factory just like we do with the transports?